resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -502,25 +512,18 @@ private[spark] class Client(
     val maybeKeyStore = sparkConf.get(KUBERNETES_DRIVER_SUBMIT_KEYSTORE)
     val resolvedSparkConf = sparkConf.clone()
     val (isLocalKeyStore, resolvedKeyStore) = maybeKeyStore.map(keyStore => {
-      val keyStoreURI = Utils.resolveURI(keyStore)
-      val isProvidedKeyStoreLocal = keyStoreURI.getScheme match {
-        case "file" | null => true
-        case "container" => false
-        case _ => throw new SparkException(s"Invalid KeyStore URI $keyStore; keyStore URI" +
-          " for submit server must have scheme file:// or container:// (no scheme defaults" +
-          " to file://)")
-      }
-      (isProvidedKeyStoreLocal, Option.apply(keyStoreURI.getPath))
+      (KubernetesFileUtils.isUriLocalFile(keyStore),
+        Option.apply(Utils.resolveURI(keyStore).getPath))
     }).getOrElse((true, Option.empty[String]))
     resolvedKeyStore.foreach {
       resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_KEYSTORE, _)
     }
     sparkConf.get(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE).foreach { trustStore =>
-      val trustStoreURI = Utils.resolveURI(trustStore)
-      trustStoreURI.getScheme match {
-        case "file" | null =>
-          resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE, trustStoreURI.getPath)
-        case _ => throw new SparkException(s"Invalid trustStore URI $trustStore; trustStore URI" +
+      if (KubernetesFileUtils.isUriLocalFile(trustStore)) {
+        resolvedSparkConf.set(KUBERNETES_DRIVER_SUBMIT_TRUSTSTORE,
+          Utils.resolveURI(trustStore).getPath)
+      } else {
+        throw new SparkException(s"Invalid trustStore URI $trustStore; trustStore URI" +
```

Comment:
we could download the truststore from s3/hdfs URLs here too couldn't we?  Maybe best to leave a TODO and come back to it if requested in the future